pkg/plugin/plugin.go

```diff
@@ -754,6 +756,35 @@ func (e *AutoscaleEnforcer) Score(
 	return score, nil
 }
 
+// NormalizeScore weights scores uniformly in the range [minScore, trueScore], where
+// minScore is framework.MinNodeScore + 1.
+func (e *AutoscaleEnforcer) NormalizeScore(
+	ctx context.Context,
+	state *framework.CycleState,
+	pod *corev1.Pod,
+	scores framework.NodeScoreList,
+) *framework.Status {
+	for _, nodeScore := range scores {
+		score := nodeScore.Score
+
+		// rand.Intn will panic if we pass in 0
+		if score == 0 {
+			continue
+		}
+
+		// This is different from framework.MinNodeScore. We use framework.MinNodeScore
+		// to indicate that a pod should not be placed on a node. The lowest
+		// actual score we assign a node is thus framework.MinNodeScore + 1
+		minScore := framework.MinNodeScore + 1
+
+		// We want to pick a score in the range [minScore, score], so use
+		// score _+ 1_ - minscore, as rand.Intn picks a number in the _half open_
+		// range [0, n)
+		nodeScore.Score = int64(rand.Intn(int(score+1-minScore))) + minScore
+	}
+	return nil
+}
+
 // ScoreExtensions is required for framework.ScorePlugin, and can return nil if it's not used
 func (e *AutoscaleEnforcer) ScoreExtensions() framework.ScoreExtensions {
```

Comment:
IIRC you also need to return `*AutoscaleEnforcer` from this function in order for the NormalizeScore method to get used

Reply:
Are you sure? I don't see anything here: https://pkg.go.dev/k8s.io/kubernetes/pkg/scheduler/framework#ScoreExtensions

Reply:
I might be wrong, but my intuition is that given the ScoreExtensions method exists (and especially because there's no corresponding `NormalizeScorePlugin` interface), the scheduler won't call the NormalizeScore method without you telling it to.

You should be able to tell either way with logs though ;)

Reply:
Ohhh I see. I interpreted "this function" as the new one, instead of the `ScoreExtensions` one. Should've paid closer attention to the section of code you selected. Indeed, the docs say:
```
// ScoreExtensions returns a ScoreExtensions interface if it implements one, or nil if does not.
ScoreExtensions() ScoreExtensions
```

Reply:
Fixed in 72b53c3a0193e812ea9d19aee2d39ce750ae54ed